Makefile

```diff
@@ -29,19 +29,21 @@ clean: ## remove generated byte code, coverage reports, and build artifacts
 
 upgrade: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
 	pip install -q pip-tools
+	pip-compile --allow-unsafe --rebuild -o requirements/pip.txt requirements/pip.in
 	pip-compile --rebuild --upgrade -o requirements/base.txt requirements/base.in
 	pip-compile --rebuild --upgrade -o requirements/dev.txt requirements/dev.in requirements/quality.in
 	pip-compile --rebuild --upgrade -o requirements/doc.txt requirements/base.in requirements/doc.in
 	pip-compile --rebuild --upgrade -o requirements/quality.txt requirements/quality.in
 	pip-compile --rebuild --upgrade -o requirements/test.txt requirements/base.in requirements/test.in
-	pip-compile --rebuild --upgrade -o requirements/travis.txt requirements/travis.in
+	pip-compile --rebuild --upgrade -o requirements/ci.txt requirements/ci.in
 	cat requirements/dev-extra.in >> requirements/dev.txt
-	cat requirements/dev-extra.in >> requirements/travis.txt
+	cat requirements/dev-extra.in >> requirements/ci.txt
```

Comment:
This `dev-extra.in` thing is weird, and adding unpinned dependencies to our requirements files.  Could you look into this a bit to see if we still need it, and if so find a better way to handle this?